src/utils/forms-loader.ts

```diff
@@ -169,27 +197,43 @@ export function applyFormIntent(intent: string, originalJson) {
 
 function updateQuestionRequiredBehaviour(question, intent) {
   const requiredIntentBehaviour = question.behaviours?.find(behaviour => behaviour.intent === intent);
-  // If required intent is present, substitute original props
-  if (requiredIntentBehaviour) {
-    question.required = requiredIntentBehaviour.required || undefined;
-    question.unspecified = requiredIntentBehaviour.unspecified || undefined;
-    question.hide = requiredIntentBehaviour.hide || undefined;
-    question.validators = requiredIntentBehaviour.validators || undefined;
-    question.value = requiredIntentBehaviour.value || undefined;
-  } else {
-    // Attempt to retrieve default behaviours
-    const defaultIntentBehaviour = question.behaviours?.find(behaviour => behaviour.intent === '*');
-    if (defaultIntentBehaviour) {
-      question.required = defaultIntentBehaviour.required || undefined;
-      question.unspecified = defaultIntentBehaviour.unspecified || undefined;
-      question.hide = defaultIntentBehaviour.hide || undefined;
-      question.validators = defaultIntentBehaviour.validators || undefined;
-      question.value = defaultIntentBehaviour.value || undefined;
-    }
-  }
 
-  // make sure behaviours prop is always deleted
-  if (question.behaviours) {
+  const defaultIntentBehaviour = question.behaviours?.find(bevahiour => bevahiour.intent === '*');
+
+  // If both required and default intents exist, combine them and update to question
+  if (requiredIntentBehaviour && defaultIntentBehaviour) {
+    // Remove the intent name props from each object
+    delete requiredIntentBehaviour.intent;
+    delete defaultIntentBehaviour.intent;
+
+    // Combine required and default intents following the rules:
+    // 1. The default intent is applied to all other intents
+    // 2. Intent-specific behaviour overrides default behaviour
+    const combinedBehaviours = Object.assign(defaultIntentBehaviour, requiredIntentBehaviour);
+
+    // Add the combinedBehaviours data to initial question
+    question = Object.assign(question, combinedBehaviours);
+
+    // Remove behaviours list
     delete question.behaviours;
   }
 }
+
+function updateMarkdownRequiredBehaviour(markdown, intent) {
+  const requiredIntentBehaviour = markdown.behaviours?.find(behaviour => behaviour.intent === intent);
+  const defaultIntentBehaviour = markdown.behaviours?.find(behaviour => behaviour.intent === '*');
+
+  if (requiredIntentBehaviour && defaultIntentBehaviour) {
+    delete requiredIntentBehaviour.intent;
+    delete defaultIntentBehaviour.intent;
+    const combinedBehaviours = Object.assign(defaultIntentBehaviour, requiredIntentBehaviour);
+
+    markdown = Object.assign(markdown, combinedBehaviours);
+    delete markdown.behaviours;
+  } else if (!requiredIntentBehaviour && defaultIntentBehaviour) {
+    delete defaultIntentBehaviour.intent;
+
+    markdown = Object.assign(markdown, defaultIntentBehaviour);
+    delete markdown.behaviours;
+  }
```

Comment:
@larslemos why do you need this? If markdowns are questions, then you don't need this

Reply:
Ist an additional pre-processor, we shouldn't bloat the code for each feature pre-processor.